src/database.rs

```diff
@@ -513,6 +516,31 @@ impl Database {
         self.verify.as_ref().unwrap()
     }
 
+    /// Get a statement by label.
+    pub fn statement(&mut self, name: &str) -> Option<StatementRef> {
+        match self.name_result().lookup_label(name.as_bytes()) {
+            None => None,
+            Some(lookup) => Some(self.parse_result().statement(lookup.address)),
+        }
+    }
+
+    /// Export an mmp file for a given statement.
+    pub fn export(&mut self, stmt: String) {
+        time(&self.options.clone(), "export", || {
+            let parse = self.parse_result().clone();
```

Comment:
You mean like array bounds checks?  I want dependent types too, let's have a static constraint that the `VarIndex`es in a `Frame` are all less than the `mandatory_count`, that would remove some instructions that are executed over a million times each :)

I'm not super concerned about this because (a) the instructions in question would be executed less than ten times each, (b) `RefCell` isn't necessary for `Arc` in the first place: https://github.com/rust-lang/rfcs/pull/1651 (all that `Cell` actually requires is that the value can be copied without running user code.  `Copy` provides that, but so do `Rc` and `Arc`)